app/Http/Controllers/UserController.php

```diff
@@ -103,4 +106,103 @@ public function getUsersByLocation(Location $location): \Illuminate\Http\Resourc
     {
         return UserResource::collection(User::all()->where('location_id', $location->id));
     }
+
+    /**
+     * Register user
+     *
+     * @param UserRequest $request
+     * @return JsonResponse
+     */
+    public function register(UserRequest $request): JsonResponse
+    {
+        // create the user
+        $user = User::create([
+            'first_name' => $request->input('first_name'),
+            'last_name' => $request->input('last_name'),
+            'phone_number' => $request->input('phone_number'),
+            'email' => $request->input('email'),
+            'password' => Hash::make($request->input('password')),
+            'location_id' => $request->input('location_id'),
+            'role_id' => 1
+        ]);
+
+        return $this->login($request);
+    }
+
+    /**
+     * Login user
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function login(Request $request): JsonResponse
+    {
+        if (!\Auth::attempt($request->only('email', 'password'))) {
+            return response()->json([
+                'message' => 'Login information is invalid.'
+            ], 401);
+        }
+
+        // get user
+        $user = User::where('email', $request['email'])->firstOrFail();
+
+        // get current time and add an hour
+        $expirationsDate = Carbon::now()->addHour();
+
+        // create abilities based on user role
+        $abilities = $user->role->role_abilities->pluck('name')->toArray();
+
+        // create token with the given abilities
+        $token = $user->createToken('authToken', $abilities, $expirationsDate)->plainTextToken;
+
+        return response()->json([
+            'access_token' => $token,
+            'token_type' => 'Bearer',
+        ]);
+    }
+
+    /**
+     * Logout user
+     *
+     * @return void
+     */
+    public function logout(): void
+    {
+        if (auth('sanctum')->check()) {
+            auth('sanctum')->user()->tokens()->delete();
+        }
+    }
+
+    /**
+     * Check users token for ability
+     *
+     * @param Request $request
+     * @param $ability
+     * @return JsonResponse
+     */
+    public function checkAbility($ability): JsonResponse
+    {
+        if (auth('sanctum')->check()) {
+            //return $request->user('sanctum')->currentAccessToken()->tokenCan($ability);
+
+            $user = auth('sanctum')->user();
+            $token = $user->tokens()->first();
+            $abilities = $token->abilities;
+
+            // return $ability;
+            // return $abilities;
+            if (in_array($ability, $abilities)) {
+                return response()->json([
+                    'has_ability' => true
+                ]);
+            }
+            return response()->json([
+                'has_ability' => false
+            ]);
+        } else {
+            return response()->json([
+                'message' => 'Unauthenticated',
+            ], 403);
+        }
+    }
```

Comment:
I think these out commented returns should be removed.

Reply:
Unused code removed and replaced with minor documentation